### ISSUE TITLE
Hub pull-request doesn't take GitHub repository settings into account

### ISSUE THREAD

**Issue by aensidhe** [github/hub#814]
Hello.

We have a private repo with default branch named `develop`. All pull requests go there and `master` branch is for stable code. But hub doesn't take this into account and tries to open pull requests to master if I do not use `-b` option.

Hub version:
````
$ hub version
git version 1.9.4.msysgit.2
hub version 2.2.0
````

**Comment by mislav** [comment]
Hub might not be able to determine which is your main branch and chooses "master" by default. Can you try this and see if it helps:

```
git remote set-head origin develop
```

More detail in e645f9aaed46ea6c00f64a54fae92b1d68e81dbb

**Comment by aensidhe** [comment]
Yep, it worked. Thank you. Maybe it's worth to add this issue to some FAQ or manual?

**Comment by mislav** [comment]
Maybe. But git should have set this automatically when you first cloned the repo locally (it should have detected the default branch). Is it possible that you first cloned the repo and only later defined "develop" branch as default one?

**Comment by aensidhe** [comment]
Yeah, it happened right as you described. We adopted git-flow after starting of development, so yeah, repo was cloned month ago and default branch was changed only a week ago.

**Comment by mislav** [comment]
I'm not sure how we can fix this. For newly cloned repos, the default branch setting will be detected. For repos cloned before the default branch change, this issue will arise. But I'm not sure if it's worth fixing. I'll leave this open to solicit opinions and similar reports.

**Comment by henri-hulski** [comment]
Same problem here. Maybe add a command to set default branch.

**Comment by henri-hulski** [comment]
Or better just add the possibility of setting default branch with `git remote set-head origin defaultbranch` to the help text of `-b` flag.
